Content.Server/Botany/Systems/MutationSystem.cs

```diff
@@ -138,94 +140,97 @@ public SeedData Cross(SeedData a, SeedData b)
     // one bit gets flipped.
     private void MutateFloat(ref float val, float min, float max, int bits, int totalbits, float mult)
     {
-        // Probability that a bit flip happens for this value.
-        float p = mult*bits/totalbits;
-        p = Math.Clamp(p, 0, 1);
-        if (!Random(p))
+        // Probability that a bit flip happens for this value's representation in thermometer code.
+        float probBitflip = mult * bits / totalbits;
+        probBitflip = Math.Clamp(probBitflip, 0, 1);
+        if (!Random(probBitflip))
         {
             return;
         }
 
         // Starting number of bits that are high, between 0 and bits.
-        int n = (int)Math.Round((val - min) / (max - min) * bits);
-        // val may be outside the range of min/max due to starting prototype values, so clamp
-        n = Math.Clamp(n, 0, bits);
+        // In other words, it's val mapped linearly from range [min, max] to range [0, bits], and then rounded.
+        int valInt = (int) MathF.Round((val - min) / (max - min) * bits);
```

Comment:
Convention is *not* to detach the cast from the thing it's casting, see: https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/types/casting-and-type-conversions

Reply:
Done.